helpdesk_mgmt/controllers/main.py

```diff
@@ -12,8 +14,64 @@ def support_ticket_close(self, **kw):
         values = {}
         for field_name, field_value in kw.items():
             values[field_name] = field_value
-        ticket = http.request.env['helpdesk.ticket'].sudo().\
-            search([('id', '=', values['ticket_id'])])
-        ticket.stage_id = values.get('stage_id')
+        ticket = http.request.env['helpdesk.ticket'].search([(
+            'id', '=', values['ticket_id'])])
+        stage_id = int(values.get('stage_id'))
+        ticket.sudo().update(
+            {'stage_id': stage_id})
 
         return werkzeug.utils.redirect("/my/ticket/" + str(ticket.id))
+
+    @http.route('/new/ticket', type="http", auth="user", website=True)
+    def create_new_ticket(self, **kw):
+        categories = http.request.env['helpdesk.ticket.category']. \
+            search([('active', '=', True)])
+        email = http.request.env.user.email
+        name = http.request.env.user.name
+
+        return http.request.render('helpdesk_mgmt.portal_create_ticket', {
+            'categories': categories, 'email': email, 'name': name})
+
+    @http.route('/submitted/ticket',
+                type="http", auth="user", website=True, csrf=True)
+    def submit_ticket(self, **kw):
+        vals = {'company_id': request.env.user.company_id.id}
+
+        for field_name, field_value in kw.items():
```

Comment:
This is an open door for injecting ticket values. Can you limit this? How is done in other places in Odoo?